pkg/webui/console/views/application-integration-edit/index.js

```diff
@@ -0,0 +1,159 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { Component } from 'react'
+import { Container, Col, Row } from 'react-grid-system'
+import bind from 'autobind-decorator'
+import { connect } from 'react-redux'
+import { defineMessages } from 'react-intl'
+import { replace } from 'connected-react-router'
+
+import Breadcrumb from '../../../components/breadcrumbs/breadcrumb'
+import { withBreadcrumb } from '../../../components/breadcrumbs/context'
+import IntlHelmet from '../../../lib/components/intl-helmet'
+import Message from '../../../lib/components/message'
+import WebhookForm from '../../components/webhook-form'
+import toast from '../../../components/toast'
+import Spinner from '../../../components/spinner'
+
+import sharedMessages from '../../../lib/shared-messages'
+
+import { getWebhookId } from '../../../lib/selectors/id'
+import { webhookSelector, fetchingSelector, errorSelector } from '../../store/selectors/webhook'
+import { selectSelectedApplicationId } from '../../store/selectors/application'
+import { getWebhook } from '../../store/actions/webhook'
+import diff from '../../../lib/diff'
+
+import api from '../../api'
+
+const m = defineMessages({
+  editWebhook: 'Edit Webhook',
+  updateSuccess: 'Successfully updated webhook',
+  deleteSuccess: 'Successfully deleted webhook',
+})
+
+const webhookEntitySelector = [
+  'base_url',
+  'format',
+  'uplink_message',
+  'join_accept',
+  'downlink_ack',
+  'downlink_nack',
+  'downlink_sent',
+  'downlink_failed',
+  'downlink_queued',
+  'location_solved',
+]
+
+@connect(function (state) {
+  const webhook = webhookSelector(state)
+  return {
+    appId: selectSelectedApplicationId(state),
+    webhook,
+    whkId: getWebhookId(webhook),
+    fetching: fetchingSelector(state),
+    error: errorSelector(state),
+  }
+}, function (dispatch, { match }) {
+  const { appId, webhookId } = match.params
+  return {
+    getWebhook: () => dispatch(getWebhook(appId, webhookId, webhookEntitySelector)),
+    dispatch,
+  }
+})
+@withBreadcrumb('apps.single.integrations.edit', function (props) {
+  const { appId, match } = props
+  const { webhookId } = match.params
```

Comment:
you can get it from the `props`

Reply:
I think I'll rather get it only from `match.params`, as the mapped `webhookId` value is not immediately available.